05-tasks-and-events.md

```diff
@@ -34,85 +32,140 @@ Tasks MUST follow those rules:
 Required parameters are:
 * `hash`: Transaction hash.
-* `confirmations`: Confirmation threshold to wait for before producing an event. If 0, the event is created as soon as a transaction is seen.
 * `confirmation-bound`: Upper bound on the confirmation count until which the syncer should report updates on the block depth of the transaction. This task MUST be maintained until this threshold is reached or until until `lifetime` has passed.
 * `lifetime`: Epoch at which the syncer SHOULD drop this task. This task MUST be maintained until this threshold is reached or until until `confirmation-bound` has been reached. 
 
-Once a transaction is seen by the syncer, and passed the confirmation threshold, a `transaction_seen` event is emitted. It contains:
-* `confirmations`: Current confirmation threshold.
-Further fields may be defined depending on the coin. Any coin based on a blockchain MUST also have:
-* `block`: The hash of the block which contains this transaction. If the transaction has yet to be included in a block, a zero value hash is used.
+Once a transaction is seen by the syncer, and passed the confirmation threshold, a `transaction_confirmations` event is emitted. 
 
 ### The `broadcast_transaction` Task
 
-`broadcast_transaction` tells a syncer to broadcast a transaction. The syncer MUST broadcast the transaction, even if it was already broadcasted.
+`broadcast_transaction` tells a syncer to broadcast a transaction. The syncer MUST broadcast the transaction, unless the transaction already in the blockchain or in the mempool.
 
 The only parameter is:
 * `tx`: The raw transaction in its serialized format.
 
+The blockchain event in response is `transaction broadcasted` event.
+
 ## Blockchain Events
+The function of the Syncer is to emit events to accomplish its assigned tasks. Syncer's must emit events targeting the daemon (at a later stage of the project, potentially the client as well). Blockchain Events are produced by syncers in response to certain type of `tasks`. A task MAY produce multiple `blockchain event` messages. The `blockchain event` messages as defined below. 
+
+### The `task_aborted` Event
+Event in response to `abort_task` Task, emitted only once.
+Data:
+* `id`: integer
+* `success_abort`: bool
+
+### The `height_changed` Event
+`watch_height` task was previously defined. That task emits `height changed` blockchain events. Upon new block arrival and reorgs, a `height_changed` event is emitted. It contains:
+* `block`: current block_hash 
+* `height`: current blockchain height.
+
+Upon task reception, syncers MUST send a `height_changed` event immediately to signify the current height.
+
+### The `address_transaction` Event
+Once a `watch_address` address is involved in a transaction, `address_transaction`  emitted. It contains:
+* `hash`: Transaction ID.
+* `amount`: Value of the amount sent to the specified address.
+
+Further fields may be defined depending on the coin. Any coin based on a blockchain MUST also have:
```

Comment:
Does 'asset based' better than 'coin based'. I think I used it quite a lot lately on the RFC and code Doc